doc/generate_cet.py

```diff
@@ -3,7 +3,121 @@
 import pathlib
 
 
+def provide_ignored():
+    ignored_attributes = ["EFDB_Name", "Description"]
+    ignored_fields = ["Description", "Alias", "Device", "Property", "Action", "Value", "Subsystem", "Version", "Author", "Explanation"]
+    return ignored_attributes, ignored_fields
+
+def add_generics(cf, subsystem, set_name, has_generics, has_specific):
+    ignored_attributes, ignored_fields = provide_ignored()
+    if has_generics is True:
+        gen_tree = ET.parse("../sal_interfaces/SALGenerics.xml")
+        gen_root = gen_tree.getroot()
+        for gen_set in gen_root:
+            gen_set[:] = sorted(gen_set, key=lambda child: (child.tag, child.find("EFDB_Topic").text.split("_")[-1] if child.find("EFDB_Topic") is not None else child.tag))
+            gen_set_name = gen_set.tag
+            if gen_set_name == set_name:
+                if gen_set_name[3:-3] not in has_specific:
+                    cf.write(f"{gen_set_name[3:-3]}s\n")
```

Comment:
Also where does the variable `cf` come from? You write to it, but it is not in the argument list. I very strongly suggest passing all values as arguments (or using a class and putting the value in "self"). Global variables make code much harder to maintain.